parsing/g_constr.mlg

```diff
@@ -244,11 +244,11 @@ GRAMMAR EXTEND Gram
       ] ]
   ;
   record_declaration:
-    [ [ fs = record_fields -> { CAst.make ~loc @@ CRecord fs } ] ]
+    [ [ fs = term_record_fields -> { CAst.make ~loc @@ CRecord fs } ] ]
   ;
 
-  record_fields:
-    [ [ f = record_field_declaration; ";"; fs = record_fields -> { f :: fs }
+  term_record_fields:
+    [ [ f = record_field_declaration; ";"; fs = term_record_fields -> { f :: fs }
```

Comment:
@jfehrle: I think we can do better at unifying the syntax of records in `g_constr.mlg` and of records in `g_vernac.mlg`. I propose to do it as part of #11103 as follows:

`G_constr`:
- `record_fields` -> `record_fields_instance`
- `record_field` -> `record_field_instance`

`G_vernac`:
- `record_fields` kept unchanged

The motivation is that `record_fields` in `G_constr` is actually the syntax for the inhabitant of a record type, while in `G_vernac` it is about the record type itself.

Reply:
Done, though `record_field` -> `record_field_instance` is in g_vernac.